si-id-epu50/si_id_epu50/epu.py

```diff
@@ -1139,39 +1150,21 @@ def polarization_motion(self, start: bool = False) -> None:
 
         # phase movement
         self.phase_start(True)
-        self.polarization = _cte.polarization_mon.index('none')
+        self.polarization = self._pol_none
         while self.phase_is_moving:
             time.sleep(0.2)
 
         # finalize polarization movement
         self.pol_is_moving = False
 
-    @staticmethod
-    def is_equal_with_tolerance(
-            value: float, reference: float, tolerance: float = 0):
-        """Check if two values are equals with a given tolerance."""
-        return abs(value - reference) <= tolerance
-
-    def update_polarization_status(self):
-        """Update polarization property."""
-        # check if polarization is defined
-        pol_phases = _cte.pol_phases
-        for pol_idx in pol_phases.keys():
-            if self.is_equal_with_tolerance(
-                    self.phase, pol_phases[pol_idx], Epu.PPARAM_TOL):
-                self.polarization = pol_idx
-                return
-
-        # checking if changing polarization
-        if self.is_equal_with_tolerance(
-                    self.gap, _cte.id_parked_gap, Epu.KPARAM_TOL):
-                pol_idx = _cte.polarization_mon.index('none')
-                self.polarization = pol_idx
-                return
-
-        # at this point the configuration must be undefined
-        pol_idx = _cte.polarization_mon.index('undef')
-        self.polarization = pol_idx
+    def update_polarization_status(self) -> int:
+        """ Polarization property. """
+        idname = self.args.pv_prefix
+        pol_idx = _IDSearch.conv_idname_2_polarization_state(
+            idname, pparameter=self.phase, kparameter=self.gap)
+        if pol_idx != self._pol_undef:
+            self.polarization = pol_idx
+        return pol_idx
```

Comment:
it seems the return type changed, the function used to return None

Reply:
yes. we used this int return in a code version we tried in a machine study. but it is not necessary anymore.

Reply:
done